clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/orchestration/DefaultOrchestrationProcessor.groovy

```diff
@@ -24,15 +24,28 @@ import groovy.util.logging.Slf4j
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.context.ApplicationContext
 
+import java.lang.reflect.Field
 import java.util.concurrent.ExecutorService
-import java.util.concurrent.Executors
+import java.util.concurrent.SynchronousQueue
+import java.util.concurrent.ThreadPoolExecutor
+import java.util.concurrent.TimeUnit
 import java.util.concurrent.TimeoutException
 
+import static com.netflix.spinnaker.security.AuthenticatedRequest.propagate
+
 @Slf4j
 class DefaultOrchestrationProcessor implements OrchestrationProcessor {
   private static final String TASK_PHASE = "ORCHESTRATION"
 
-  protected ExecutorService executorService = Executors.newCachedThreadPool()
+  protected ExecutorService executorService = new ThreadPoolExecutor(0, Integer.MAX_VALUE,
+    60L, TimeUnit.SECONDS,
```

Comment:
Same thing that `Executions.newCachedThreadPool()` does but with the added `afterExecute()` that should clear the thread locals.